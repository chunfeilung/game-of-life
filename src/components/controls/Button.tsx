import * as React from "react";

export interface IButtonProps {
  id: string;
  label: string;
  onClick: () => void;
}

export const Button = (props: IButtonProps) => {
  return (
    <button id={props.id} onClick={props.onClick}>{props.label}</button>
  );
};
