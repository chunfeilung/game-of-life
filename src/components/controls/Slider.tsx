import * as React from "react";

export interface ISliderProps {
  id: string;
  value: number;
  onChange: (value: number) => void;
}

export interface ISliderState {
  value: number;
}

export class Slider extends React.Component<ISliderProps, ISliderState> {
  constructor(props: ISliderProps) {
    super(props);
    this.state = {value: props.value};

    this.handleChange = this.handleChange.bind(this);
  }

  public render = () => {
    return (
      <label>
        Speed:
        <input
          id={this.props.id}
          type="range"
          min="1"
          max="10"
          value={this.state.value}
          onChange={this.handleChange}
        />
      </label>
    );
  }

  private handleChange = (event: any) => {
    this.setState({value: event.target.value});
    this.props.onChange(event.target.value);
  }
}
