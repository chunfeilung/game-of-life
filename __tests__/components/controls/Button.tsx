import * as React from "react";
import * as Renderer from "react-test-renderer";
import {Button} from "../../../src/components/controls/Button";

it("executes functions when clicked", () => {
  /* Given */
  let isFunctionExecuted = false;

  const handleOnClick = () => isFunctionExecuted = true;

  const component = Renderer.create(
    <Button id="testButton" label="Push me" onClick={handleOnClick}/>,
  );

  /* When */
  const snapshot = component.toJSON();
  snapshot.props.onClick();

  /* Then */
  expect(snapshot).toMatchSnapshot();
  expect(isFunctionExecuted).toBe(true);
});
