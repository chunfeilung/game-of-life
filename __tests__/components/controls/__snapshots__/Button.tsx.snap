// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`executes functions when clicked 1`] = `
<button
  id="testButton"
  onClick={[Function]}
>
  Push me
</button>
`;
